content/en/docs/languages/php/context.md

```diff
@@ -0,0 +1,181 @@
+
+// active context is unchanged
+var_dump(Context::getCurrent()->get($key)); // NULL
+```
+
+If a value is not found in the current context, then each parent is checked
+until either the key is found, or the root context is reached.
+
+## Active Context
+
+The active context is the context which is returned by `Context::getCurrent()`.
+The context object contains entries which allow telemetry components to
+communicate with each other. For example, when a span is created it may be
+activated, which creates a new active context and stores the span. Later, when
+another span is created it may use the span from the active context as its
+parent span. If no context is active, the root context is returned, which is
+just the empty context object.
+
+```php
+use OpenTelemetry\Context\Context;
+
+// Returns the active context
+// If no context is active, the root context is returned
+$context = Context::getCurrent();
+```
+
+### Set Active Context
+
+A context can be made active by calling `$context->activate()`.
+
+```php
+use OpenTelemetry\Context\Context;
+
+$key = Context::createKey('my-key');
+$ctx = Context::getCurrent();
+$ctx2 = $ctx->with($key, 'context 2');
+$ctx2->activate();
+assert($ctx2 === Context::getCurrent());
+```
+
+#### Scope
+
+The return value of `$context->activate()` is a `Scope`. You must `detach()` the
+scope to "de-activate" that context, which will make the previously-active
+context the active context again.
+
+The return value of `$scope->detach()` is an integer. A zero return value means
+that the scope was successfully detached. A non-zero value means that the call
+was unexpected; either the context associated with the scope was:
+
+- already detached
+- not a part of the current execution context
+- not the active context
+
+#### DebugScope
+
+To assist developers in locating issues with context and scope, there is
+`DebugScope`. In development (that is, a PHP runtime with assertions enabled),
+an activated `Context` will be wrapped in a `DebugScope`. The `DebugScope` keeps
+track of when the scope was activated, and has a destructor which triggers an
+error if the scope was not detached. The error output contains a backtrace of
+which code activated the context.
+
+The following code would trigger an error, complaining that a scope was not
+detached, and giving a backtrace of where the scope was created:
+
+```php
+use OpenTelemetry\Context\Context;
+
+$key = Context::createKey('my-key');
+$scope = Context::getCurrent()->with($key, 'value')->activate();
+
+//exit without detaching $scope
+```
+
+This can be problematic in some situations, particularly in legacy applications
+which might `exit` or `die` - any active spans will not be completed and
+exported, but the `DebugScope` will also complain loudly.
+
+If you understand why `DebugScope` is complaining and accept the risks, then you
+can disable the feature entirely by setting `OTEL_PHP_DEBUG_SCOPES_DISABLED` to
+a truthy value.
```

Comment:
```suggestion
This can be problematic in some situations, particularly in legacy applications
which might `exit` or `die`. In that case, active spans are not completed and
exported, and the `DebugScope` complains loudly.

If you understand why `DebugScope` is complaining and accept the risks, then you
can disable the feature entirely by setting `OTEL_PHP_DEBUG_SCOPES_DISABLED` to
a truthy value.
```

Can we be more specific as to what "complains loudly" means or is noticed? Perhaps a log sample?